forge/routes/api/deviceLive.js

```diff
@@ -38,6 +38,13 @@ module.exports = async function (app) {
             })
             return
         }
+        if (request.body.settingsHash !== (request.device.settingsHash)) {
```

Comment:
We should only do this check if `request.body` includes `settingsHash` - otherwise skip the check. Otherwise we'll break existing agents that will get the 409 but assume its snapshot related.